server/graphql/v2/mutation/VendorMutations.ts

```diff
@@ -0,0 +1,145 @@
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../input/AccountReferenceInput';
+import { GraphQLVendorCreateInput, GraphQLVendorEditInput } from '../input/VendorInput';
+import { GraphQLVendor } from '../object/Vendor';
+
+const VENDOR_INFO_FIELDS = ['contact', 'taxFormUrl', 'taxType', 'taxId', 'notes'];
+
+const vendorMutations = {
+  createVendor: {
+    type: new GraphQLNonNull(GraphQLVendor),
+    description: 'Create a new vendor for given host',
+    args: {
+      host: {
+        description: 'Reference to the host that holds the vendor',
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+      },
+      vendor: {
+        description: 'The vendor to create',
+        type: new GraphQLNonNull(GraphQLVendorCreateInput),
+      },
+    },
+    resolve: async (_, args, req) => {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be authenticated to perform this action. Please login and try again.');
+      }
+
+      const host = await fetchAccountWithReference(args.host, { loaders: req.loaders, throwIfMissing: true });
+      if (!req.remoteUser.isAdminOfCollective(host)) {
+        throw new Unauthorized("You're not authorized to create a vendor for this host");
+      }
+
+      const { vendorInfo } = args.vendor;
+      const vendorData = {
+        type: CollectiveType.VENDOR,
+        slug: `${host.id}-${slugify(args.vendor.name)}-${uuid().substr(0, 8)}`,
+        CreatedByUserId: req.remoteUser.id,
+        image: args.vendor.imageUrl || null,
+        isActive: false,
+        ParentCollectiveId: host.id,
+        ...pick(args.vendor, ['name', 'legalName', 'tags']),
+        data: {
+          vendorInfo: pick(vendorInfo, VENDOR_INFO_FIELDS),
+        },
+      };
+
+      if (vendorInfo.taxType) {
+        assert(vendorInfo.taxId, new BadRequest('taxId is required when taxType is provided'));
+        vendorData.data[vendorInfo.taxType] = vendorInfo.taxId;
+      }
+
+      const vendor = await models.Collective.create(vendorData);
+
+      if (args.vendor.address) {
+        await vendor.setLocation({ address: args.vendor.address });
+      }
+
+      return vendor;
+    },
+  },
+  editVendor: {
+    type: new GraphQLNonNull(GraphQLVendor),
+    description: 'Edit an existing vendor',
+    args: {
+      vendor: {
+        description: 'Reference to the host that holds the vendor',
+        type: new GraphQLNonNull(GraphQLVendorEditInput),
+      },
+    },
+    resolve: async (_, args, req) => {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be authenticated to perform this action. Please login and try again.');
+      }
+
+      const vendor = await fetchAccountWithReference(args.vendor, { loaders: req.loaders, throwIfMissing: true });
+      const host = await req.loaders.Collective.byId.load(vendor.ParentCollectiveId);
+      assert(host, new NotFound('Vendor host not found'));
```

Comment:
We should check the account type

```suggestion
      assert(host, new NotFound('Vendor host not found'));
      assert(vendor.type === 'VENDOR', new ValidationFailed('Account is not a vendor'));
```